### ISSUE TITLE
FastQC KeyError for reports with zero reads

### ISSUE THREAD

**Issue by ec-ho-ra-mos** [ewels/MultiQC#1533]
### Description of bug

I am running MultiQC (from [DockerHub](https://hub.docker.com/r/ewels/multiqc)) version 1.11. For some reason, I am getting the error "Sample had zero reads". Am I doing something wrong? My input files are 2 FastQC output folders (test run). Each output folder contains the following: icons and images for the 2 folders, and fastqc.fo, fastqc_data.txt, fastqc_report.html, and summary.txt for the files.

### File that triggers the error

[fastqc_data.txt](https://github.com/ewels/MultiQC/files/7007745/fastqc_data.txt)

### MultiQC Error log

My run commands are just:

`multiqc <directory>`

This is the error message:

```console
/// MultiQC ðŸ” | v1.11

| multiqc | Search path : 15_fastqc

| fastqc | Found 2 reports

| fastqc | Sample had zero reads: 'F1-1A_S1_R1_001'

| fastqc | Sample had zero reads: 'F1-1A_S1_R2_001'

â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ Oops! The 'fastqc' MultiQC module broke... â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®

â”‚ Please copy this log and report it at â”‚

â”‚ https://github.com/ewels/MultiQC/issuesâ”‚

â”‚ Please attach a file that triggers the error. The last file found was: 15_f â”‚

â”‚ astqc/F1-1A_S1_R2_001_fastqc/fastqc_data.txt â”‚

â”‚ â”‚

â”‚ Traceback (most recent call last): â”‚

â”‚ File "/usr/lib/python3.8/site-packages/multiqc/multiqc.py", line 624, in r â”‚

â”‚ output = mod() â”‚

â”‚ File "/usr/lib/python3.8/site-packages/multiqc/modules/fastqc/fastqc.py", â”‚

â”‚ self.sequence_content_plot() â”‚

â”‚ File "/usr/lib/python3.8/site-packages/multiqc/modules/fastqc/fastqc.py", â”‚

â”‚ for b in data[s_name]: â”‚

â”‚ KeyError: 'F1-1A_S1_R1_001' â”‚

â”‚ â”‚

â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯

| multiqc | No analysis results found. Cleaning up..

| multiqc | MultiQC complete
```

**Comment by ewels** [comment]
Hi @ec-ho-ra-mos,

Finally got to this PR to take a look. I owe you an apology - when I ran your supplied report it's clear that your sample _did_ have reads. This was a nasty subtle bug where the read counts were being overwritten because MultiQC assumed some values would be there. I guess you ran FastQC with several sections skipped: those missing values were incorrectly interpreted as zero reads and a bunch of values overwritten with zeros.

Anyway, long story short - should now be fixed in #1535. Let me know how you get on with it.

Thanks for reporting!

Phil

**Comment by ec-ho-ra-mos** [comment]
Dear Dr. Ewels,

Thank you very much for your update regarding this issue. I forgot to send
an update on my question but I have managed to solve my issue already. It
turned out the HPC that I was using was an older version of FastQC. What I
did was I installed the newest version of FastQC and the outputs were all
compatible with the then-newest version of MultiQC.

Thank you very much for your attention to this matter.

Sincerely,
Jherico GERONCA

------------------------------------------------
*Jherico GERONCA (Mr.)*
Research Student
Department of Molecular and Cellular Analysis
Graduate School of Pharmaceutical Sciences
Kyoto University
------------------------------------------------

[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
10/11/21,
07:45:30 am

**Comment by ewels** [comment]
Ok great - well either way, you discovered a genuine bug even if it was an edge case :) MultiQC should now work properly with the old version of FastQC as well as the newer version.
